### PR TITLE
[53509]feature search screen

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task 53510: Create UI](https://edu-redmine.sun-asterisk.vn/issues/53510)
- [#Task 53511: Handle event](https://edu-redmine.sun-asterisk.vn/issues/53511)
- [#Task 53512: get data](https://edu-redmine.sun-asterisk.vn/issues/53512)
## WHAT
- 
## Evidence (Screenshot or Video)
image
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

**Review by ghost — APPROVED** [review]

**Review by minhnn-1279 — APPROVED** [review]

**Review by ndt0409 — APPROVED** [review]

File: app/src/main/java/com/example/finalapplication/screen/historycontact/HistoryContactFragment.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package com.example.finalapplication.screen.historycontact
+
+import android.content.Intent
+import com.example.finalapplication.utils.base.BaseFragment
+import com.example.finalapplication.databinding.FragmentHistoryContactBinding
+import com.example.finalapplication.screen.search.SearchActivity
+
+class HistoryContactFragment :
+    BaseFragment<FragmentHistoryContactBinding>(FragmentHistoryContactBinding::inflate) {
+    override fun handleEvent() {
```
Comment (vietnh-1109) [inline app/src/main/java/com/example/finalapplication/screen/historycontact/HistoryContactFragment.kt]: cách ra 1 dòng

File: app/src/main/java/com/example/finalapplication/Utils/base/BaseViewHolder.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.example.finalapplication.utils.base
+
+import androidx.recyclerview.widget.RecyclerView
+import androidx.viewbinding.ViewBinding
+
+abstract class BaseViewHolder<T>(private val binding: ViewBinding) :
+    RecyclerView.ViewHolder(binding.root) {
+    abstract fun bind(item: T)
```
Comment (vietnh-1109) [inline app/src/main/java/com/example/finalapplication/Utils/base/BaseViewHolder.kt]: cách 1 dòng

File: app/src/main/java/com/example/finalapplication/screen/search/SearchActivity.kt [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    private var isEndPage = false
+    private var isLoading = false
+    private var currenquery: String? = null
+    private lateinit var searchListAdapter: SearchListAdapter
+
+    override fun handleEvent() {
+        binding.apply {
+            buttonBack.setOnClickListener { finish() }
+            searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+                override fun onQueryTextSubmit(p0: String?): Boolean {
```
Comment (vietnh-1109) [inline app/src/main/java/com/example/finalapplication/screen/search/SearchActivity.kt]: nếu biến p0 đc sử dụng thì đặt lại tên nhé

File: app/src/main/java/com/example/finalapplication/screen/search/SearchActivity.kt [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+                override fun isEndPage() = isEndPage
+            })
+        }
+    }
+
+    override fun initData() {
+        binding.apply {
+            searchView.requestFocus()
+        }
+        searchListAdapter = SearchListAdapter(this)
```
Comment (vietnh-1109) [inline app/src/main/java/com/example/finalapplication/screen/search/SearchActivity.kt]: khai báo đơn giản thế này thì vứt lên chỗ init global bên trên luôn

File: app/src/main/java/com/example/finalapplication/screen/search/SearchActivity.kt [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+            users.addAll(data)
+            searchListAdapter.submitList(users)
+            searchListAdapter.notifyDataSetChanged()
+        }
+        searchViewModel.isLoading.observe(this) { data ->
+            isLoading = data
+            if (data == true) {
+                binding.containerLoading.visibility = View.VISIBLE
+                binding.progressLoading.visibility = View.VISIBLE
+                binding.textNodata.visibility = View.GONE
```
Comment (vietnh-1109) [inline app/src/main/java/com/example/finalapplication/screen/search/SearchActivity.kt]: dùng isVisible

File: app/src/main/java/com/example/finalapplication/screen/search/SearchListAdapter.kt [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    inner class LoadViewHolder(val binding: ItemLoadMoreBinding) : BaseViewHolder<Any>(binding) {
+
+        override fun bind(item: Any) {
+            // TODO no-ip
+        }
+    }
+
+    inner class UserViewHolder(val binding: ItemUserBinding) : BaseViewHolder<Any>(binding) {
```
Comment (vietnh-1109) [inline app/src/main/java/com/example/finalapplication/screen/search/SearchListAdapter.kt:45]:
```suggestion
    inner class UserViewHolder(val binding: ItemUserBinding) : BaseViewHolder<User>(binding) {
```

File: app/src/main/java/com/example/finalapplication/screen/search/SearchViewModel.kt [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+            onRequest = {
+                userRepository.getUserByName(name, lastIndex, object : Listenner<List<User>> {
+                    override fun onSuccess(data: List<User>) {
+                        _users.value = data
+                        hideLoading(true)
+                    }
+
+                    override fun onError(msg: String) {
+                        message.value = msg
+                        _users.value = mutableListOf()
```
Comment (vietnh-1109) [inline app/src/main/java/com/example/finalapplication/screen/search/SearchViewModel.kt]: cái này để làm gì nhỉ

**Pull request merged**